node/controllers/Login-Controller.coffee

```diff
@@ -87,6 +89,74 @@ class Login_Controller
         else
             @.res.send(JSON.stringify(response));
 
+  passwordResetToken : ()=>
+    #Parsing URL
+    url = @.req?.url?.split('/')
+    username = url[2]?.toString()
+    token    = url[3]?.toString()
+
+    passwordStrengthRegularExpression =///(
+        (?=.*\d)            # at least 1 digit
+        (?=.*[A-Z])         # at least 1 upper case letter
+        (?=.*[a-z])         # at least 1 lower case letter
+        (?=.*\W)            # at least 1 special character
+        .                   # match any with previous validations
+        {8,256}             # 8 to 256 characters
+       )///
+
+    #Validating token
+    if (token == null || username == null )
+      @res.render(password_reset_fail, {errorMessage: 'Token is invalid'})
+      return
+
+    #Password not provided
+    if (@.req.body.password.length==0)
+      @res.render(password_reset_fail, {errorMessage: 'Password must not be empty'})
+      return
+
+    #Confirmation password not provided
+    if (@.req.body['confirm-password'].length==0)
+      @res.render(password_reset_fail, {errorMessage: 'Confirmation Password must not be empty'})
+      return
+
+    #Passwords must match
+    if (@.req.body.password != @.req.body['confirm-password'])
+      @res.render(password_reset_fail, {errorMessage: 'Passwords don\'t match'})
+      return
+    #length check
+    if (@.req.body.password.length < 8 || @.req.body.password.length>256 )
+      @res.render(password_reset_fail, {errorMessage: 'Password must be 8 to 256 character long'})
+      return
+
+    #Complexity
+    if (!@.req.body.password.match(passwordStrengthRegularExpression))
+      @res.render(password_reset_fail, {errorMessage: 'Your password should be at least 8 characters long. It should have one uppercase and one lowercase letter, a number and a special character'})
+      return
```

Comment:
This is ok for now, but this sequence of if statements is perfect to be solved with a more functional type of programming (which js and coffee support)